consensus/polybft/state_test.go

```diff
@@ -0,0 +1,658 @@
+	t.Helper()
+
+	for i := uint64(1); i <= numberOfCommitments; i++ {
+		commitment, err := createTestCommitmentMessage(i*stateSyncMainBundleSize, stateSyncMainBundleSize)
+		require.NoError(t, err)
+
+		commitmentToExecute := &CommitmentToExecute{
+			SignedCommitment: commitment,
+			ToIndex:          commitment.Message.ToIndex,
+		}
+		require.NoError(t, state.insertCommitmentMessage(commitmentToExecute))
+	}
+}
+
+func createTestStateSyncs(numberOfEvents, startIndex uint64) []*StateSyncEvent {
+	stateSyncEvents := make([]*StateSyncEvent, 0)
+	for i := startIndex; i < numberOfEvents+startIndex; i++ {
+		stateSyncEvents = append(stateSyncEvents, &StateSyncEvent{
+			ID:     i,
+			Sender: ethgo.ZeroAddress,
+			Target: ethgo.ZeroAddress,
+			Data:   []byte{0, 1},
+		})
+	}
+
+	return stateSyncEvents
+}
+
+func createTestCommitmentMessage(fromIndex, bundleSize uint64) (*CommitmentMessageSigned, error) {
+	// TODO - uncomment once changes from the integration with v3-contracts get moved to edge
+	// tree, err := NewMerkleTree([][]byte{
+	//      {0, 1},
+	//      {2, 3},
+	//      {4, 5},
+	// })
+	// if err != nil {
+	//      return nil, err
+	// }
+	msg := &CommitmentMessage{
+		// MerkleRootHash: tree.Hash(),
+		MerkleRootHash: types.EmptyRootHash,
+		FromIndex:      fromIndex,
+		ToIndex:        fromIndex + bundleSize - 1,
+		BundleSize:     bundleSize,
+	}
+
+	return &CommitmentMessageSigned{
+		Message:      msg,
+		AggSignature: Signature{},
+	}, nil
+}
+
+func TestMemdb_InsertGetDeleteStateSyncEvents(t *testing.T) {
+	const stateSyncCount = 10
+
+	memdb, err := memdb.NewMemDB(memStateSchema)
+	require.NoError(t, err)
+
+	stateSyncs := generateStateSyncEvents(t, stateSyncCount, 0)
+
+	for _, sse := range stateSyncs {
+		require.NoError(t, insertToMemDB(memdb, stateSyncTable, sse))
+	}
+
+	t.Run("Get all state sync events", func(t *testing.T) {
+		events, err := getFilteredFromMemDB[*StateSyncEvent](memdb, stateSyncTable, uint64(0), nil)
+		require.NoError(t, err)
+		require.Len(t, events, stateSyncCount)
+	})
+
+	t.Run("Get state sync events from beginning to middle", func(t *testing.T) {
+		events, err := getFilteredFromMemDB[*StateSyncEvent](memdb, stateSyncTable, uint64(0), uint64(4))
+
+		require.NoError(t, err)
+		require.Len(t, events, 5)
+	})
+
+	t.Run("Get state sync events from middle to end", func(t *testing.T) {
+		events, err := getFilteredFromMemDB[*StateSyncEvent](memdb, stateSyncTable, uint64(5), uint64(stateSyncCount)-1)
+
+		require.NoError(t, err)
+		require.Len(t, events, 5)
+	})
+
+	t.Run("Delete state sync events lower or equal to index 4", func(t *testing.T) {
+		err := deleteFilteredFromMemDB(memdb, stateSyncTable, uint64(4))
+		require.NoError(t, err)
+		events, err := getFilteredFromMemDB[*StateSyncEvent](memdb, stateSyncTable, uint64(0), nil)
+		require.NoError(t, err)
+		require.Len(t, events, 5)
+	})
+}
+
+// TODO - uncomment once changes from the integration with v3-contracts get moved to edge
```

Comment:
Do we need this commented out code?

Reply:
No. This test was added when we integrated with v3-contracts. I'll remove it for now, and we can add it once we integrate those changes to edge as well.

Fix: https://github.com/0xPolygon/polygon-edge/pull/785/commits/a01ff43ddb674d1acd647a8355061503ae8b202f

Reply:
I believe we need then a task to have negative tests also. How do you think?

Reply:
What do you think if I add some negative cases through this PR? It won't take me long to write them.

Reply:
Added some new negative cases. There were some already implemented, but I added those which were missing:
https://github.com/0xPolygon/polygon-edge/pull/785/commits/6ef13e95425c38452637ec904bc1a0029ae5d334